src/base58.cpp

```diff
@@ -31,11 +41,9 @@ bool DecodeBase58(const char* psz, std::vector<unsigned char>& vch)
     // Process the characters.
     while (*psz && !isspace(*psz)) {
         // Decode base58 character
-        const char* ch = strchr(pszBase58, *psz);
-        if (ch == nullptr)
+        int carry = mapBase58[(uint8_t)*psz];
```

Comment:
Should you check the character `*psz` is less than 128?

Reply:
Oops...I was thought size of `mapBase58` is 256, thanks

Reply:
Fixed, as per @laanwj suggestion `mapBase58` has 256 elements.